async-nats/src/options.rs

```diff
@@ -476,6 +479,22 @@ impl ConnectOptions {
         self.sender_capacity = capacity;
         self
     }
+
+    /// Sets custom prefix instead of default `_INBOX`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), async_nats::Error> {
+    /// async_nats::ConnectOptions::new().inbox_prefix("CUSTOM").connect("demo.nats.io").await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn inbox_prefix<T: ToString>(mut self, prefix: T) -> ConnectOptions {
```

Comment:
@caspervonb for options, maybe it's the best interface to use?

Explicitly states it will convert to owned String and yet allows for using `&str`.